src/v3/src/components/Widget/index.tsx

```diff
@@ -96,6 +96,7 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
   const [loading, setLoading] = useState<boolean>(false);
   const [widgetRendered, setWidgetRendered] = useState<boolean>(false);
   const brandedTheme = mapMuiThemeFromBrand(brandColors);
+  const [userIdentifier, setUserIdentifier] = useState<string | undefined>(undefined);
```

Comment:
To go along with the comment below, if we still want to allow non-string values `null | undefined` I favor using `null` when the value is intentionally empty.

Another way to go is _only_ allow `string` and check for empty strings in the component code before appending the parameters to the URL.